File: 백준/Silver/14916. 거스름돈/거스름돈.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int N = sc.nextInt();
		int cnt = 0;
		if(N%5==0) cnt+= N/5;
		
		else if((N%5)%2==0) cnt+= N/5 + (N%5)/2;
		
		else if ((N%5)%2==1 && N/5 != 0) cnt+= N/5 -1 + ((N%5)+5)/2;
		
		else cnt = -1;
		
		System.out.println(cnt);
	}

}

File: 백준/Silver/14916. 거스름돈/거스름돈.py
money = int(input())
cnt = 0
if money==0:
    print(-1)
elif money %5 ==0:
    print(money//5)
elif money%5 ==1 or money%5==3:
    if money//5 >= 1:
        print(money//5 - 1 + (money % 5 + 5) // 2)
    else:
        print(-1)
elif (money%5) % 2 == 0:
    print(money//5 + (money % 5) //2)

elif money%2 == 0:
    print(money//2)


